plasmapy/utils/decorators/checks.py

```diff
@@ -1,20 +1,1219 @@
 """
 Decorator for checking input/output arguments of functions.
 """
-__all__ = ["check_quantity", "check_relativistic"]
+__all__ = ["check_values", "check_units", "check_quantity", "check_relativistic",
+           "CheckBase", "CheckUnits", "CheckValues"]
 
+import collections
 import functools
 import inspect
 import numpy as np
 import warnings
 
 from astropy import units as u
-from astropy.units import UnitsWarning
 from astropy.constants import c
+from astropy.units import UnitsWarning
+from functools import reduce
 from plasmapy.utils.decorators import preserve_signature
-from plasmapy.utils.exceptions import RelativityWarning, RelativityError
+from plasmapy.utils.exceptions import (PlasmaPyWarning,
+                                       RelativityWarning,
+                                       RelativityError)
 from textwrap import dedent
-from typing import (Dict, Union)
+from typing import (Any, Dict, List, Tuple, Union)
+
+try:
+    from astropy.units.equivalencies import Equivalency
+except ImportError:
+    # TODO: remove once we have dependency Astropy >= 3.2.1
+    # astropy defined the Equivalency class in v3.2.1
```

Comment:
Oh, I didn't know that Astropy did this!  I like it.

Reply:
Yeah, I forget how I stumbled across this.  I'm glad they have, but they haven't quite propagated the class throughout AstroPy.  The old-styple equivalencies were a list of 2-4 element tuples and the new-style class sub-classes `UserList`.  Thus, the new-style plays nicely with the old foundation and the old foundation hasn't been specifically modified for the `Equivalencies` class.